fog/sample-paykit/src/cached_tx_data/mod.rs

```diff
@@ -127,10 +136,28 @@ impl CachedTxData {
     /// then we would have to track if we did that. Then this `min`
     /// expression would also take the min of outstanding missed blocks.
     pub fn get_num_blocks(&self) -> BlockCount {
-        min(
-            self.rng_set.get_highest_processed_block_count(),
-            self.key_image_data_completeness,
-        )
```

Comment:
I feel like it would be slightly easier to read this if we did it in two steps maybe:

```
let missing_block_limit = if self.missed_block_ranges.is_empty() {
    BlockCount::MAX()
} else {
    BlockCount::from(self.missed_Block_ranges.iter().map(...).min().unwrap() + 1;
};
min(
    self.rng_set.get_highest_processed_block_count(),
    self.key_image_data_completeness,
    missing_block_limit
)
```

what do you think about that?

Reply:
I think this can be further simplified to remove the if statement with:
```
let missing_block_limit = self.missed_block_ranges
                        .iter()
                        .map(|block_range| block_range.start_block + 1)
                        .min()
                        .unwrap_or_else(BlockCount::MAX);
```